### ISSUE TITLE
Floki does not cope well with containers with custom entry points

### ISSUE THREAD

**Issue by AdmiralGT** [Metaswitch/floki#163]
I have the following config file
`image: prom/prometheus:v2.22.1
shell: /bin/sh
`

The prometheus container has a custom entry point which results in floki being unable to start this container

`developer@PC6114:~/gitlab/platform.core/simon-simpl-k8s$ floki -c floki/validate_alerts.yaml
Error parsing commandline arguments: unexpected /bin/sh
prometheus: error: unexpected /bin/sh
10:52:25 [ERROR] A problem occured: Running container failed: docker run exited with return code 1`

However, just running docker exec works fine by setting the entrypoint

`developer@PC6114:~$ docker run -it --entrypoint /bin/sh -v $PWD:/src -w /src prom/prometheus:v2.22.1
/src $`

**Comment by maxdymond** [comment]
You can use `docker_switches` to make this work:
```
image: prom/prometheus:v2.22.1
shell: /bin/sh
docker_switches:
  - --entrypoint=""
```

**Comment by rlupton20** [comment]
Hi Geoff,

I agree this is a ways from ideal :-). We were discussing in the floki slack channel the possibility of making this configurable (or making it easy to suppress the entrypoint), but then got to wondering if there is a floki use-case with non-standard entrypoint. If not, it might be worth force suppressing the entrypoint by default. We could still make it configurable, but it's unclear how valuable that would be.

**Comment by maxdymond** [comment]
Has this been closed by #164 ?

**Comment by rlupton20** [comment]
I think so, yes. I'll close.
